src/RecursiveIteratorAggregateIterator.php

```diff
@@ -0,0 +1,80 @@
+<?php
+
+declare(strict_types=1);
+
+namespace loophp\iterators;
+
+use Generator;
+use Iterator;
+use IteratorAggregate;
+use RuntimeException;
+use Traversable;
+
+use function count;
+
+/**
+ * @template TKey
+ * @template T
+ *
+ * @implements IteratorAggregate<int, T>
+ */
+class RecursiveIteratorAggregateIterator implements IteratorAggregate
+{
+    /**
+     * @var list<Iterator>
+     */
+    private array $stack = [];
+
+    /**
+     * @param Traversable<TKey,T> $input
+     */
+    public function __construct(Traversable $input)
+    {
+        $this->stack[] = self::findIterator($input);
```

Comment:
I'm wondering if this statement shouldn't be moved inside `getIterator()`. Is there any benefit into doing this in the constructor?

Reply:
we need to store input somewhere before getIterator invokation

Reply:
Let's just do save that `$input` in the constructor, and process it in `getIterator()` then?

Reply:
sure